File: examples/github-client/sdks/c/src/services.c
#include "services.h"
#include <stdlib.h>
#include <string.h>

app_response_t* app_git_hub_api_get_repository(app_client_t* client, const app_request_opts_t* opts) {
    app_request_opts_t local_opts = {0};
    if (opts) local_opts = *opts;
    local_opts.method = "GET";
    local_opts.endpoint = "repos/{owner}/{repo}";
    return app_client_request(client, &local_opts);
}

app_response_t* app_git_hub_api_get_repository_commits(app_client_t* client, const app_request_opts_t* opts) {
    app_request_opts_t local_opts = {0};
    if (opts) local_opts = *opts;
    local_opts.method = "GET";
    local_opts.endpoint = "repos/{owner}/{repo}/commits";
    return app_client_request(client, &local_opts);
}

app_response_t* app_git_hub_api_get_repository_contributors(app_client_t* client, const app_request_opts_t* opts) {
    app_request_opts_t local_opts = {0};
    if (opts) local_opts = *opts;
    local_opts.method = "GET";
    local_opts.endpoint = "repos/{owner}/{repo}/contributors";
    return app_client_request(client, &local_opts);
}

app_response_t* app_git_hub_api_get_repository_issues(app_client_t* client, const app_request_opts_t* opts) {
    app_request_opts_t local_opts = {0};
    if (opts) local_opts = *opts;
    local_opts.method = "GET";
    local_opts.endpoint = "repos/{owner}/{repo}/issues";
    return app_client_request(client, &local_opts);
}

app_response_t* app_git_hub_api_get_repository_readme(app_client_t* client, const app_request_opts_t* opts) {
    app_request_opts_t local_opts = {0};
    if (opts) local_opts = *opts;
    local_opts.method = "GET";
    local_opts.endpoint = "repos/{owner}/{repo}/readme";
    return app_client_request(client, &local_opts);
}

app_response_t* app_git_hub_api_get_user(app_client_t* client, const app_request_opts_t* opts) {
    app_request_opts_t local_opts = {0};
    if (opts) local_opts = *opts;
    local_opts.method = "GET";
    local_opts.endpoint = "users/{username}";
    return app_client_request(client, &local_opts);
}

app_response_t* app_git_hub_api_get_user_repositories(app_client_t* client, const app_request_opts_t* opts) {
    app_request_opts_t local_opts = {0};
    if (opts) local_opts = *opts;
    local_opts.method = "GET";
    local_opts.endpoint = "users/{username}/repos";
    return app_client_request(client, &local_opts);
}

app_response_t* app_git_hub_api_get_user_starred_repositories(app_client_t* client, const app_request_opts_t* opts) {
    app_request_opts_t local_opts = {0};
    if (opts) local_opts = *opts;
    local_opts.method = "GET";
    local_opts.endpoint = "users/{username}/starred";
    return app_client_request(client, &local_opts);
}

app_response_t* app_git_hub_api_search_code(app_client_t* client, const app_request_opts_t* opts) {
    app_request_opts_t local_opts = {0};
    if (opts) local_opts = *opts;
    local_opts.method = "GET";
    local_opts.endpoint = "search/code";
    return app_client_request(client, &local_opts);
}

app_response_t* app_git_hub_api_search_repositories(app_client_t* client, const app_request_opts_t* opts) {
    app_request_opts_t local_opts = {0};
    if (opts) local_opts = *opts;
    local_opts.method = "GET";
    local_opts.endpoint = "search/repositories";
    return app_client_request(client, &local_opts);
}

app_response_t* app_git_hub_api_search_users(app_client_t* client, const app_request_opts_t* opts) {
    app_request_opts_t local_opts = {0};
    if (opts) local_opts = *opts;
    local_opts.method = "GET";
    local_opts.endpoint = "search/users";
    return app_client_request(client, &local_opts);
}

File: examples/github-client/sdks/rust/src/services.rs
use crate::client::Client;
use reqwest::header::HeaderMap;
use std::collections::HashMap;

/// Service for GitHubApiService containing 11 endpoints
pub struct GitHubApiService;

impl GitHubApiService {
    /// GET repos/{owner}/{repo}
    pub async fn get_repository(
        client: &Client,
        owner: &str,
        repo: &str,
    ) -> Result<reqwest::Response, reqwest::Error> {
        let mut path_params = HashMap::new();
        path_params.insert("owner", owner);
        path_params.insert("repo", repo);
        let pp_arg = Some(&path_params);
        client.send_request(
            reqwest::Method::GET,
            "repos/{owner}/{repo}",
            pp_arg,
            None,
            None,
            None,
        ).await
    }

    /// GET repos/{owner}/{repo}/commits
    pub async fn get_repository_commits(
        client: &Client,
        owner: &str,
        repo: &str,
        query_params: Option<&HashMap<&str, &str>>,
    ) -> Result<reqwest::Response, reqwest::Error> {
        let mut path_params = HashMap::new();
        path_params.insert("owner", owner);
        path_params.insert("repo", repo);
        let pp_arg = Some(&path_params);
        client.send_request(
            reqwest::Method::GET,
            "repos/{owner}/{repo}/commits",
            pp_arg,
            query_params,
            None,
            None,
        ).await
    }

    /// GET repos/{owner}/{repo}/contributors
    pub async fn get_repository_contributors(
        client: &Client,
        owner: &str,
        repo: &str,
        query_params: Option<&HashMap<&str, &str>>,
    ) -> Result<reqwest::Response, reqwest::Error> {
        let mut path_params = HashMap::new();
        path_params.insert("owner", owner);
        path_params.insert("repo", repo);
        let pp_arg = Some(&path_params);
        client.send_request(
            reqwest::Method::GET,
            "repos/{owner}/{repo}/contributors",
            pp_arg,
            query_params,
            None,
            None,
        ).await
    }

    /// GET repos/{owner}/{repo}/issues
    pub async fn get_repository_issues(
        client: &Client,
        owner: &str,
        repo: &str,
        query_params: Option<&HashMap<&str, &str>>,
    ) -> Result<reqwest::Response, reqwest::Error> {
        let mut path_params = HashMap::new();
        path_params.insert("owner", owner);
        path_params.insert("repo", repo);
        let pp_arg = Some(&path_params);
        client.send_request(
            reqwest::Method::GET,
            "repos/{owner}/{repo}/issues",
            pp_arg,
            query_params,
            None,
            None,
        ).await
    }

    /// GET repos/{owner}/{repo}/readme
    pub async fn get_repository_readme(
        client: &Client,
        owner: &str,
        repo: &str,
    ) -> Result<reqwest::Response, reqwest::Error> {
        let mut path_params = HashMap::new();
        path_params.insert("owner", owner);
        path_params.insert("repo", repo);
        let pp_arg = Some(&path_params);
        client.send_request(
            reqwest::Method::GET,
            "repos/{owner}/{repo}/readme",
            pp_arg,
            None,
            None,
            None,
        ).await
    }

    /// GET users/{username}
    pub async fn get_user(
        client: &Client,
        username: &str,
    ) -> Result<reqwest::Response, reqwest::Error> {
        let mut path_params = HashMap::new();
        path_params.insert("username", username);
        let pp_arg = Some(&path_params);
        client.send_request(
            reqwest::Method::GET,
            "users/{username}",
            pp_arg,
            None,
            None,
            None,
        ).await
    }

    /// GET users/{username}/repos
    pub async fn get_user_repositories(
        client: &Client,
        username: &str,
        query_params: Option<&HashMap<&str, &str>>,
    ) -> Result<reqwest::Response, reqwest::Error> {
        let mut path_params = HashMap::new();
        path_params.insert("username", username);
        let pp_arg = Some(&path_params);
        client.send_request(
            reqwest::Method::GET,
            "users/{username}/repos",
            pp_arg,
            query_params,
            None,
            None,
        ).await
    }

    /// GET users/{username}/starred
    pub async fn get_user_starred_repositories(
        client: &Client,
        username: &str,
        query_params: Option<&HashMap<&str, &str>>,
    ) -> Result<reqwest::Response, reqwest::Error> {
        let mut path_params = HashMap::new();
        path_params.insert("username", username);
        let pp_arg = Some(&path_params);
        client.send_request(
            reqwest::Method::GET,
            "users/{username}/starred",
            pp_arg,
            query_params,
            None,
            None,
        ).await
    }

    /// GET search/code
    pub async fn search_code(
        client: &Client,
        query_params: Option<&HashMap<&str, &str>>,
    ) -> Result<reqwest::Response, reqwest::Error> {
        let pp_arg = None;
        client.send_request(
            reqwest::Method::GET,
            "search/code",
            pp_arg,
            query_params,
            None,
            None,
        ).await
    }

    /// GET search/repositories
    pub async fn search_repositories(
        client: &Client,
        query_params: Option<&HashMap<&str, &str>>,
    ) -> Result<reqwest::Response, reqwest::Error> {
        let pp_arg = None;
        client.send_request(
            reqwest::Method::GET,
            "search/repositories",
            pp_arg,
            query_params,
            None,
            None,
        ).await
    }

    /// GET search/users
    pub async fn search_users(
        client: &Client,
        query_params: Option<&HashMap<&str, &str>>,
    ) -> Result<reqwest::Response, reqwest::Error> {
        let pp_arg = None;
        client.send_request(
            reqwest::Method::GET,
            "search/users",
            pp_arg,
            query_params,
            None,
            None,
        ).await
    }

}
